// @brightleaf/react-hooks/lib/use-axios.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.useAxios = void 0;

var _react = require("react");

var _axios = _interopRequireDefault(require("axios"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const reducer = (state, action) => {
  switch (action.type) {
    case 'get':
      return { ...state,
        loading: true
      };

    case 'success':
      return { ...state,
        data: action.payload.data,
        error: null,
        loading: false
      };

    case 'error':
      return { ...state,
        data: null,
        error: action.payload.error,
        loading: false
      };

    default:
      return state;
  }
};

const useAxios = url => {
  const [state, dispatch] = (0, _react.useReducer)(reducer, {
    data: [],
    error: null,
    loading: true
  });

  const fetchQuery = async url => {
    dispatch({
      type: 'get'
    });
    const resp = await _axios.default.get(url);
    const data = resp.data;
    dispatch({
      type: 'success',
      payload: {
        data
      }
    });
  };

  (0, _react.useEffect)(() => {
    fetchQuery(url);
  }, []);
  return state;
};

exports.useAxios = useAxios;
var _default = useAxios;
exports.default = _default;
