// @brightleaf/react-hooks/lib/use-post.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.usePost = void 0;

var _useRequest = _interopRequireDefault(require("./use-request"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

const defaultConfig = {
  mode: 'cors',
  cache: 'no-cache',
  credentials: 'same-origin',
  headers: {
    'Content-Type': 'application/json'
  },
  redirect: 'follow',
  referrer: 'no-referrer'
  /**
   * @typedef {Object} PostRequest~State
   * @property {Object|array} data - The data returned from the function.
   * @property {boolean} error - Indicates whether the function had an error.
   * @property {boolean} loading - Indicates whether the function is executing
   * @property {function} postData - The function to call to make the POST request
   */

  /**
   * usePost hook to handle posting data to a url
   *
   * @param {string} url - The url to request
   * @param {object} config - The config object to be used with the fetch
   * @returns {...PostRequest~State} - The states and results and the call to make the request
   */

};

const usePost = (url = '', config = {
  method: 'POST'
}) => {
  const {
    data,
    error,
    loading,
    makeRequest
  } = (0, _useRequest.default)(url, {
    data: [],
    error: null,
    loading: false
  });
  const fullConfig = { ...defaultConfig,
    ...config
  };
  /**
   * postData - Post data to url
   *
   * @param {object} data - the data to post
   */

  const postData = async data => {
    await makeRequest({
      method: 'POST',
      ...fullConfig,
      body: JSON.stringify(data)
    });
  };

  return {
    data,
    error,
    loading,
    postData
  };
};

exports.usePost = usePost;
var _default = usePost;
exports.default = _default;
