// @brightleaf/react-hooks/lib/use-mutation.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.useMutation = void 0;

var _graphqlRequest = require("graphql-request");

var _react = require("react");

const reducer = (state, action) => {
  switch (action.type) {
    case 'get':
      return { ...state,
        loading: true
      };

    case 'success':
      return { ...state,
        data: action.payload.data,
        error: null,
        loading: false
      };

    case 'error':
      return { ...state,
        data: null,
        error: action.payload.error,
        loading: false
      };

    default:
      return state;
  }
};

const useMutation = function (url, query) {
  const [state, dispatch] = (0, _react.useReducer)(reducer, {
    data: [],
    error: null,
    loading: true
  });

  const makeQuery = async variables => {
    dispatch({
      type: 'get'
    });
    const resp = await (0, _graphqlRequest.request)(url, query, variables);
    const data = resp;
    dispatch({
      type: 'success',
      payload: {
        data
      }
    });
  };

  return { ...state,
    makeQuery
  };
};

exports.useMutation = useMutation;
var _default = useMutation;
exports.default = _default;
